protocol/contact.go

```diff
@@ -191,10 +179,45 @@ func (c *Contact) mutual() bool {
 	return c.added() && c.hasAddedUs()
 }
 
+func (c *Contact) active() bool {
+	return c.mutual() && !c.Blocked
+}
+
 func (c *Contact) dismissed() bool {
 	return c.ContactRequestLocalState == ContactRequestStateDismissed
 }
 
+func (c *Contact) names() []string {
+	var names []string
+
+	if c.LocalNickname != "" {
+		names = append(names, c.LocalNickname)
+	}
+
+	if c.ENSVerified && len(c.EnsName) != 0 {
```

Comment:
hey @cammellos sorry i didn't review properly, we should have secondary name only if we have nickname, if we don't have it, secondary name should be empty, so there is no need in names array, we should set only two field, 
```clojure
(defn assoc-names
  [{:keys [nickname display-name ens-name ens-verified alias public-key] :as contact}]
  (let [username (cond ens-verified
                       ens-name
                       (not (string/blank? display-name))
                       display-name
                       :else
                       (or alias (gfycat/generate-gfy public-key)))]
    (if (not (string/blank? nickname))
      (assoc contact :first-name nickname :second-name username)
      (assoc contact :first-name username))))
```

Reply:
oh I see, I'll change that then